### PR TITLE
GBE-180: warn on overbook no matter how a volunteer is booked

### PR DESCRIPTION
**Pull request by bethlakshmi**
- creates the same 2 warnings regardless of how the volunteer was booked
 - refactored the code to remove the VolunteerBidChangeForm - which was weird and blurred the lines between gbe and scheduler
- added tests for all cases

runcoverage is 91%, 600 lines of untested code.

### CODE REVIEW

File: expo/gbe/forms.py [hunk 1]
```diff
@@ -328,97 +327,6 @@ class Meta:
         help_texts = acceptance_help_texts
 
 
-class EventCheckBox(ModelMultipleChoiceField):
```
Comment (jonkiparsky) [inline expo/gbe/forms.py]: It's always nice to see code going away.

**Comment by jonkiparsky** [comment]
This looks okay to me - no code smell, tests pass. However:
```
vagrant@blueprint:/vagrant/expo$ pepdiff
/vagrant /vagrant/expo
expo/expo/settings.py:141:37: W291 trailing whitespace
expo/expo/settings.py:238:8: E121 continuation line under-indented for hanging indent
expo/expo/settings.py:239:11: E121 continuation line under-indented for hanging indent
expo/gbe/cms_plugins.py:8:5: E123 closing bracket does not match indentation of opening bracket's line
expo/gbe/forms.py:62:25: E126 continuation line over-indented for hanging indent
expo/gbe/forms.py:66:25: E126 continuation line over-indented for hanging indent
expo/gbe/models/remains.py:317:80: E501 line too long (84 > 79 characters)
expo/gbe/models/remains.py:1085:17: E123 closing bracket does not match indentation of opening bracket's line
expo/gbe/views/volunteer_changestate_view.py:41:25: E126 continuation line over-indented for hanging indent
expo/scheduler/views.py:180:26: E126 continuation line over-indented for hanging indent
expo/scheduler/views.py:683:33: E126 continuation line over-indented for hanging indent
expo/tests/contexts/staff_area_context.py:51:53: E226 missing whitespace around arithmetic operator
expo/tests/gbe/test_act_changestate.py:64:60: E226 missing whitespace around arithmetic operator
expo/tests/gbe/test_gbe_cms_plugins.py:1:1: E902 IOError: [Errno 2] No such file or directory: 'expo/tests/gbe/test_gbe_cms_plugins.py'
/vagrant/expo
vagrant@blueprint:/vagrant/expo$ 
```

**Comment by bethlakshmi** [comment]
Thanks!  Ick - sorry about missing the pepdiffs.

**Comment by bethlakshmi** [comment]
OK, weird, after a merge from master, I can't find any of these pep8 problems... I have trouble believing that the merge could fix these...  expecially given that the merge didn't touch several areas.

@jonkiparsky - are you sure you're doing a pepdiff on this branch?

**Comment by BenignCremator** [comment]
There is this one that I do not know what to do with:
expo/gbe/models/remains.py:317:80: E501 line too long (84 > 79 characters)

Line 317 has an object name that is 68 char long.  With proper indent, it runs over the 79 char limit.  Is there a better way to deal with DB filters?

**Comment by BenignCremator** [comment]
Code looks good, tests look good, only errors were in non-relevant sections of code, and were mostly os errors, media upload dir does not exist.  Not this code.  I approve.

**Comment by BenignCremator** [comment]
Functional testing does work.  However, you need to either wait for the cache to time out and clear, or switch users to drop the cache on the floor.  Also, the pop-up warning could be larger (it is small enough to easily miss) and last longer (you can read it if you catch it immediately when it pops up).  Neither are blockers, and the functionality is there.

**Pull request merged**